### PR TITLE
qemu: Add 9p support for darwin

### PR DESCRIPTION
**Pull request by baude**
Qemu already has support for 9p in Linux.  Adding this for MacOS users
will bring them into parity for 9p support.

This replaces Ashley's #95318

Signed-off-by: Brent Baude <bbaude@redhat.com>

- [X ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Could you run `brew style --fix homebrew/core/qemu`?

**Comment by baude** [comment]
I  ran it but nothing in git appears to have changed.  Is that expected?

**Comment by SMillerDev** [comment]
Are you working in a new clone of homebrew core?

**Comment by baude** [comment]
> Are you working in a new clone of homebrew core?

@SMillerDev  Yessir.

**Comment by SMillerDev** [comment]
Yeah, that's the issue then.

You can also point `brew style --fix` to the file I believe.

The general recommendation is to work in your live brew install. 
https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request#formulae-related-pull-request

**Comment by baude** [comment]
@SMillerDev  So as a noob here, should I be doing what the docs recommend which is to get the patches to apply and then do a git diff on the source code and capture that as foo.diff?  And then foo.diff would just be included versus downloaded?

**Comment by SMillerDev** [comment]
No, the downloaded option is definitely easier. Just change the file extension you're downloading.

**Comment by baude** [comment]
We good?

**Comment by cho-m** [comment]
Linux build failure:
```
../meson.build:1386:0: ERROR: Feature virtfs cannot be enabled: virtio-9p (virtfs) on Linux requires libcap-ng-devel and libattr-devel
```

Needs Linux-only dependencies within the `on_linux do` block. Maybe:
```diff
   on_linux do
+    depends_on "attr"
     depends_on "gcc"
     depends_on "gtk+3"
+    depends_on "libcap-ng"
   end
```

File: Formula/qemu.rb [hunk 1]
```diff
@@ -36,18 +36,76 @@ class Qemu < Formula
   depends_on "vde"
 
   on_linux do
+    depends_on "attr"
     depends_on "gcc"
     depends_on "gtk+3"
+    depends_on "libcap-pg"
```
Comment (tbutter) [inline Formula/qemu.rb:42]: libcap-ng ?

Comment (SMillerDev) [reply Formula/qemu.rb:42]: See the message above

Comment (SMillerDev) [reply Formula/qemu.rb:42]: Oh nevermind, I reread it now.
```suggestion
    depends_on "libcap-ng"
```

**Review by carlocab — COMMENTED** [review]
We probably need to increment this formula's `revision`.

When making changes to this PR, please amend your existing commit instead of adding new ones. When you do so, change your commit message to
```
qemu: add 9p support for Darwin
```
(or similar) so that it conforms to the commit style guide.

File: Formula/qemu.rb [hunk 2]
```diff
@@ -36,18 +36,76 @@ class Qemu < Formula
...
+    sha256 "e2835578eeea09b75309fc3ac4a040b47c0ac8149150d8ddf45f7228ab7b5433"
+  end
+  patch do
+    url "https://raw.githubusercontent.com/baude/homebrew-qemu/798fdd7c6e2924591f45b282b3f59cb6e9850504/remove_9p-util.diff"
+    sha256 "ccf31a8e60ac7fc54fd287eca7e63fe1c9154e346d2a1367b33630227b88144d"
+  end
+  patch do
+    url "https://raw.githubusercontent.com/ashley-cui/homebrew-podman/e1162ec457bd46ed84aef9a0aa41e80787121088/change.patch"
+    sha256 "af8343144aea8b51852b8bf7c48f94082353c5e0c57d78fc61e7c3e4be3658b9"
+  end
```
[the remark this answers is not in the document]
Comment (willcohen) [reply Formula/qemu.rb:69]: Yes, however when backporting to QEMU 6.2.0, the second commit in this patch set doesn't apply correctly in brew. The version you see here is functionally the same.

File: Formula/qemu.rb [hunk 3]
```diff
@@ -36,18 +36,76 @@ class Qemu < Formula
...
+    sha256 "a349c6de07fcf8314a1d84cacc05c68573728641cc5054d0fc149d14e1c9bed8"
+  end
+  patch do
+    url "https://gitlab.com/qemu-project/qemu/-/commit/d3671fd972cd185a6923433aa4802f54d8b62112.diff"
+    sha256 "f40dd472ec4dcbf6f352338de85c1aba5a92a3b9f0a691f8ae51e298e2b5a273"
+  end
+  patch do
+    url "https://raw.githubusercontent.com/NixOS/nixpkgs/8fc669a1dd84ae0db237fdb30e84c9f47e0e9436/pkgs/applications/virtualization/qemu/allow-virtfs-on-darwin.patch"
+    sha256 "61422ab60ed9dfa3d9fe8a267c54fab230f100e9ba92275bc98cf5da9e388cde"
+  end
```
[the remark this answers is not in the document]
Comment (willcohen) [reply Formula/qemu.rb:105]: Yes, the same idea applies -- the last commit in this patch set ALSO doesn't apply cleanly when backported to 6.2.0. The version being referenced here is from the backport I prepared for nix to 6.2.0.

File: Formula/qemu.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (baude) [reply Formula/qemu.rb:69]: yes

**Comment by baude** [comment]
> We probably need to increment this formula's `revision`.

I fixed up the other issues.  Is incrementing something I do? If so, can you please elaborate?

**Comment by willcohen** [comment]
It'd just be adding `revision 1` under the head line near the top, I think!

File: Formula/qemu.rb [hunk 4]
```diff
@@ -36,18 +36,76 @@ class Qemu < Formula
...
+
+  # 820KB floppy disk image file of FreeDOS 1.2, used to test QEMU
```
[the remark this answers is not in the document]
Comment (carlocab) [reply Formula/qemu.rb:102]: Move the comment, not the `resource` block.

File: Formula/qemu.rb [hunk 5]
```diff
@@ -36,14 +36,71 @@ class Qemu < Formula
...
+    sha256 "f40dd472ec4dcbf6f352338de85c1aba5a92a3b9f0a691f8ae51e298e2b5a273"
+  end
+  patch do
+    url "https://raw.githubusercontent.com/NixOS/nixpkgs/8fc669a1dd84ae0db237fdb30e84c9f47e0e9436/pkgs/applications/virtualization/qemu/allow-virtfs-on-darwin.patch"
+    sha256 "61422ab60ed9dfa3d9fe8a267c54fab230f100e9ba92275bc98cf5da9e388cde"
+  end
+
   # 820KB floppy disk image file of FreeDOS 1.2, used to test QEMU
-  resource "homebrew-test-image" do
+    resource "homebrew-test-image" do
```
Comment (carlocab) [inline Formula/qemu.rb:103]:
```suggestion
  resource "homebrew-test-image" do
```

**Comment by baude** [comment]
What happened that two of the tasks were cancelled?

**Comment by trevor-viljoen** [comment]
LGTM

**Comment by SMillerDev** [comment]
Can you add the patches from https://github.com/Homebrew/homebrew-core/pull/96743 too?

**Comment by baude** [comment]
I've added #96743 as asked ... I think the a maintainer needs to set the long ci timeout label?

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by saschasc** [comment]
Great to have 9p support soon!

I noticed that if I compile QEMU myself and run `brew install spice-protocol` beforehand, the clipboard sharing also works afterwards when compiling it from source. Would it be possible to add this to QEMU as well? Where would be the best place to request this? Who could help?

**Pull request closed without merging**